frontend/src/Components/ContentComponent.tsx

```diff
@@ -127,8 +128,39 @@ function updateMailbox(mailbox: HTMLSpanElement, setMailboxKey: (key: MailboxKey
   })
 }
 
-function updateMail(mail: HTMLSpanElement, setMailboxKey: (key: MailKey | null) => void, currentUsername?: string) {
-  // check processed
+function renderWithTheme(container: HTMLElement, content: React.ReactNode, appState: AppState) {
+  const root = createRoot(container)
+  elementToRoot.set(container, root)
+
+  const renderContent = () => {
+    if (!elementToRoot.has(container)) return
+    root.render(<FakeRoot appState={appState}>{content}</FakeRoot>)
+  }
+
+  renderContent()
+
+  const observer = new MutationObserver(renderContent)
+  const mainThemeProvider = document.querySelector('[data-theme-provider]')
+  if (mainThemeProvider) {
+    observer.observe(mainThemeProvider, {
+      attributes: true,
+      attributeFilter: ['data-theme'],
+    })
+  }
+
+  return () => {
```

Comment:
I suggest collecting all of those from the whole tree of `updateContent` calls, and adding a cleanup on unmount in the effect of `ContentComponent` that calls `updateContent`.